.github/workflows/merge-approved-pr.yml

```
@@ -0,0 +1,132 @@
+name: Merge Approved PR
+
+on:
+  workflow_call:
+    inputs:
+      pr-number:
+        required: true
+        type: number
+        description: 'The number of the pull request to process'
+    secrets:
+      github-token:
+        required: true
+        description: 'GitHub token with permissions to merge'
+
+jobs:
+  merge-pr:
+    runs-on: ubuntu-latest
+    steps:
+      # Fetch PR metadata (head and base branches) using the GitHub API
+      - name: Get PR Details
+        id: get-pr
+        uses: actions/github-script@v7
+        with:
+          github-token: ${{ secrets.github-token }}
+          script: |
+            // Fetch full details of the pull request associated with the comment
+            const { data: pr } = await github.rest.pulls.get({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              pull_number: ${{ inputs.pr-number }}
+            });
+
+            // Output the base and head branch names for subsequent steps
+            core.setOutput('base', pr.base.ref);
+            core.setOutput('head', pr.head.ref);
+
+      # Verify that the PR targets 'main' from 'stable-main-x.y.z'
+      - name: Verify Branch Names
+        id: verify-branches
+        run: |
+          # Define the required branch pattern
+          REQUIRED_BASE="main"
+          # Head branch must match pattern: stable-main-X.Y.Z where X, Y, Z are integers
+          HEAD_PATTERN="^stable-main-[0-9]+\.[0-9]+\.[0-9]+$"
+
+          # Get actual values from the previous step
+          ACTUAL_BASE="${{ steps.get-pr.outputs.base }}"
+          ACTUAL_HEAD="${{ steps.get-pr.outputs.head }}"
+
+          # Compare actual branches against requirements
+          if [[ "$ACTUAL_BASE" != "$REQUIRED_BASE" ]] || ! [[ "$ACTUAL_HEAD" =~ $HEAD_PATTERN ]]; then
+            echo "Skipping: PR must be from 'stable-main-X.Y.Z' to '$REQUIRED_BASE'. Found $ACTUAL_HEAD -> $ACTUAL_BASE"
+            echo "should_skip=true" >> "$GITHUB_OUTPUT"
+          else
+            echo "Branches match requirements: Source '$ACTUAL_HEAD' -> Target '$ACTUAL_BASE'."
+            echo "should_skip=false" >> "$GITHUB_OUTPUT"
+          fi
+
+      # Check if the PR has the required approval status
+      - name: Verify Approval
+        id: verify-approval
+        if: success() && steps.verify-branches.outputs.should_skip != 'true'
```

The `success` check is redundant. See https://docs.github.com/en/actions/reference/workflows-and-actions/expressions#status-check-functions

>A default status check of success() is applied unless you include one of these functions


```suggestion
        if: steps.verify-branches.outputs.should_skip != 'true'
```